### ISSUE TITLE
Docs: no-restricted-imports

### ISSUE THREAD

**Issue by ArielGueta** [eslint/eslint#18454]
### Docs page(s)

https://eslint.org/docs/latest/rules/no-restricted-imports

### What documentation issue do you want to solve?

I have the following configuration:

```
            "patterns": [
              {
                "importNamePattern": "_Enum$",
                "group": ["@app/api"]
              }
            ],
```

The issue is that it also errors for "@app/api/foo" etc. How can I make it exact? 

### What do you think is the correct solution?

To include "exact" example

### Participation

- [ ] I am willing to submit a pull request for this change.

**Comment by mdjermanovic** [comment]
Hi @ArielGueta, thanks for the issue!
 
> The issue is that it also errors for "@app/api/foo" etc. How can I make it exact?

Can you clarify with a few examples of code that should and code that shouldn't error what you mean by "exact"?

If you want to restrict imports `from "@app/api"` only, you can use [`paths`](https://eslint.org/docs/latest/rules/no-restricted-imports#paths).

**Comment by ArielGueta** [comment]
Sure.

```ts
// Should be lint error
import { Foo_Enum } from '@app/api';

// These should be OK
import { Foo_Enum } from '@app/api/bar';
import { Foo_Enum } from '@app/api/baz';
```

**Comment by Tanujkanti4441** [comment]
Hi @ArielGueta, to prevent the files inside `@app/api` from being reported you can use `!` negation mark

```js
eslint no-restricted-imports: ["error", { patterns: [{
    "importNamePattern": "_Enum$",
    "group": ["@app/api", "!@app/api/"]
}]}]

// error
import { Foo_Enum } from '@app/api';

// no error
import { Foo_Enum } from '@app/api/bar';
import { Foo_Enum } from '@app/api/baz';
```
[playground](https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tcmVzdHJpY3RlZC1pbXBvcnRzOiBbXCJlcnJvclwiLCB7IHBhdHRlcm5zOiBbe1xuICAgIFwiaW1wb3J0TmFtZVBhdHRlcm5cIjogXCJfRW51bSRcIixcbiAgICBcImdyb3VwXCI6IFtcIkBhcHAvYXBpXCIsIFwiIUBhcHAvYXBpL1wiXVxufV19XSovXG5cbi8vIGVycm9yXG5pbXBvcnQgeyBGb29fRW51bSB9IGZyb20gJ0BhcHAvYXBpJztcblxuLy8gbm8gZXJyb3JcbmltcG9ydCB7IEJhcl9FbnVtIH0gZnJvbSAnQGFwcC9hcGkvYmFyJztcbmltcG9ydCB7IEJhel9FbnVtIH0gZnJvbSAnQGFwcC9hcGkvYmF6JzsiLCJvcHRpb25zIjp7InJ1bGVzIjp7fSwibGFuZ3VhZ2VPcHRpb25zIjp7InNvdXJjZVR5cGUiOiJtb2R1bGUiLCJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e319fX19)

let us know if this solves your issue!

**Comment by ArielGueta** [comment]
Works like a charm. Thanks!

It'll be valuable example to add to the docs.

**Comment by ArielGueta** [comment]
Hi,

How can I allow only one import while exclude the others. This doesn't work from some reason:

```
/*eslint no-restricted-imports: ["error", { patterns: [{
    "importNamePattern": "_Enum$",
    "group": ["@app/api", "!@app/api/enums"]
}]}]*/

// error
import { Foo_Enum } from '@app/api';
import { Bar_Enum } from '@app/api/bar';
import { Baz_Enum } from '@app/api/baz';

// no error
import { B_Enum } from '@app/api/enums';
```

**Comment by Tanujkanti4441** [comment]
To achieve this we have to write a series of patterns like this:
```js
/*eslint no-restricted-imports: ["error", { patterns: [{
    "importNamePattern": "_Enum$",
    "group": ["@app/api", "!@app/api/", "@app/api/**", "!@app/api/enums"]
}]}]*/

// error
import { Foo_Enum } from '@app/api';
import { Bar_Enum } from '@app/api/bar';
import { Baz_Enum } from '@app/api/baz';

// no error
import { B_Enum } from '@app/api/enums';
```
[playground](https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tcmVzdHJpY3RlZC1pbXBvcnRzOiBbXCJlcnJvclwiLCB7IHBhdHRlcm5zOiBbe1xuICAgIFwiaW1wb3J0TmFtZVBhdHRlcm5cIjogXCJfRW51bSRcIixcbiAgICBcImdyb3VwXCI6IFtcIkBhcHAvYXBpXCIsIFwiIUBhcHAvYXBpL1wiLCBcIkBhcHAvYXBpLyoqXCIsIFwiIUBhcHAvYXBpL2VudW1zXCJdXG59XX1dKi9cblxuLy8gZXJyb3JcbmltcG9ydCB7IEZvb19FbnVtIH0gZnJvbSAnQGFwcC9hcGknO1xuaW1wb3J0IHsgQmFyX0VudW0gfSBmcm9tICdAYXBwL2FwaS9iYXInO1xuaW1wb3J0IHsgQmF6X0VudW0gfSBmcm9tICdAYXBwL2FwaS9iYXonO1xuXG4vLyBubyBlcnJvclxuaW1wb3J0IHsgQl9FbnVtIH0gZnJvbSAnQGFwcC9hcGkvZW51bXMnOyIsIm9wdGlvbnMiOnsicnVsZXMiOnt9LCJsYW5ndWFnZU9wdGlvbnMiOnsic291cmNlVHlwZSI6Im1vZHVsZSIsInBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fX19fX0=)

**Comment by ArielGueta** [comment]
Thanks! I wish it was simple as just one regex.

**Comment by Tanujkanti4441** [comment]
> Thanks! I wish it was simple as just one regex.

You can open an issue for this idea.
